docs/infrastructure/logging.md

```diff
@@ -10,250 +10,272 @@ order: 6
 aggregation of logs from multiple components. This service can be started with
-one of the main components, allowing other components connect to it. The service
-aggregates all logs in one place for easier analysis of the interaction between
-components.
+one of the main components, allowing other components to connect to it. The
+service aggregates all logs in one place for easier analysis of the interaction
+between components. Components can also write to console, files directly without
+involving an additional logging server.
 
 <!-- MarkdownTOC levels="2,3" autolink="true" -->
 
-- [Protocol](#protocol)
-  - [Types](#types)
-  - [Messages](#messages)
-  - [Examples](#examples)
+- [Configuration](#configuration)
+  - [Custom Log Levels](#custom-log-levels)
+  - [Appenders](#appenders)
+    - [Format](#format)
+    - [File](#file-appender)
+    - [Network](#socket-appender)
+    - [Sentry.io](#sentry-appender)
 - [JVM Architecture](#jvm-architecture)
   - [SLF4J Interface](#slf4j-interface)
   - [Setting Up Logging](#setting-up-logging)
   - [Log Masking](#log-masking)
-  - [Configuration](#configuration)
   - [Logging in Tests](#logging-in-tests)
 
 <!-- /MarkdownTOC -->
 
-## Protocol
+## Configuration
 
-The service relies on a WebSocket connection to a specified endpoint that
-exchanges JSON-encoded text messages. The communication is uni-directional - the
-only messages are log messages that are sent from a connected client to the
-server that aggregates the logs.
+The logging settings should be placed under the `logging-service` key of the
+`application.conf` config. Each of the main components can customize format and
+output target via section in `application.conf` configuration file. The
+configuration is using HOCON-style, as defined by
+[lightbend/config](https://github.com/lightbend/config). Individual values
+accepted in the config are inspired by SLF4J's properties, formatting and
+implementations.
 
-### Types
+The configuration has two main sections:
 
-##### `LogLevel`
+- [custom log levels](#custom-log-levels)
+- [applications' appenders](#appenders) (also known as configuration of log
+  events output target)
 
-The log level encoded as a number. Possible values are:
+During component's setup, its `application.conf` file is read, parsed and
+verified and available as part of the class hierarchy defined in
+`org.enso.logger.config` package. Class
+`org.enso.logger.config.LoggingServiceConfig` encapsulates the complete
+information represented by the `logging-service` key of the config file and is
+used to programmatically initialize loggers.
 
-- 0 - indicating `ERROR` level,
-- 1 - indicating `WARN` level,
-- 2 - indicating `INFO` level,
-- 3 - indicating `DEBUG` level,
-- 4 - indicating `TRACE` level.
+As per [configuration schema](https://github.com/lightbend/config) any key can
+have a default value that can be overridden by an environmental variable. For
```

Comment:
```suggestion
have a default value that can be overridden by an environment variable. For
```

nitpick, but I have never heard `environmental variable` before; sounds like a term related to ecology rather than programming 😅